### DISCUSSION TITLE
`Window.Current` is `null` in Custom Protocol Activation `OnLaunch()`

### DISCUSSION THREAD

**Discussion by BertrandDeSaintBezier** [microsoft/microsoft-ui-xaml#9000] [Q&A]
Duplicate of issue in the Uno Platform repo : https://github.com/unoplatform/uno/discussions/14005#discussion-5748871

Essentially, I noticed that some things have changed in the `OnLaunch()`. It appears that Window.Current is always set to `null` when launched using [custom protocol activation](https://platform.uno/docs/articles/features/protocol-activation.html).

Is this normal ?

**Comment by ReneeGA2020** [comment]
Per the document, Window.Current should not be used in WinUI 3. It will always be null. WinUI 3 support multiple root windows anyway.

One way to figure out your Root Window is to keep a list of all your active windows. And loop through them and see if your element's XamlRoot == window.Content.XamlRoot

**Reply by BertrandDeSaintBezier** [reply]
Running through the Debugger, Windows.Current doesn't start off as null after going through OnLaunch() once, it is only when this method is called from a protocol activation that the value is null for some reason... But I will give your proposal a shot

**Comment by vash72** [comment]
@BertrandDeSaintBezier 
in WASDK it's always null, you can add a property on the App so you can get static MyApp.Current.MyMainWindow

>Desktop apps always return null for this property.

from https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.window.current?view=windows-app-sdk-1.4
